### DISCUSSION TITLE
got a "UnknownCommandError" error when using driver.contexts

### DISCUSSION THREAD

**Discussion by BianJianyu** [appium/appium#18896] [Q&A]
I use appium on m1 mac to test a hybird app for macos.
When i using "driver.contexts" to get the contexts ,it just raise a exception.

Stacktrace:
UnknownCommandError: Unhandled endpoint: /session/E4FC60DF-10E1-4838-A690-F80509282D16/contexts -- http://127.0.0.1:10100/ with parameters {
    wildcards =     (
        "session/E4FC60DF-10E1-4838-A690-F80509282D16/contexts"
    );
}

Hope somebody can help me with it!
My python code is like this:

import time

from appium import webdriver

desired_caps = {
    'automationName': 'Mac2',
    'platformName': 'mac',
    'bundleId': 'XXXX',
    'noReset': True
}

driver = webdriver.Remote('http://localhost:4723', desired_caps)
print(driver.contexts)

**Comment by KazuCocoa** [comment]
It is because mac2 driver does not support webview context

**Reply by lxmn1** [reply]
Hi, can appium automate actions in a browser from the native app? Im using Mac2Driver for automating a native app that redirects to a login screen in the Chrome browser (not in-app)? I wish to automate the filling in of fields in the login screen in the browser before switching back to the MacOS app

**Reply by KazuCocoa** [reply]
I wondered if you could start chromedriver outside the Appium session with https://github.com/appium/appium-chromium-driver or vanilla chromedriver to handle the chrome browser only.

**Reply by MagdelineNg** [reply]
Hi, thanks that worked. I successfully initialised another instance to automate the browser. 

Can mac2 driver detect a second pop-up(part of app, not browser) screen that overlays my native macos app so I can interact with the pop-up screen? I tried the code below but both give unhandled endpoint (so i believe mac2 driver does not support switching to the overlay?)

```
print(driver.window_handles)
driver.switch_to.window(driver.window_handles[-1])
driver._switch_to.context("NATIVE_APP")
print(driver.contexts)
```

^above gives 
```
Stacktrace:
UnknownCommandError: Unhandled endpoint: /session/4729C9AE-64C2-4677-B1A4-0EF6A4EAC550/window/handles -- http://127.0.0.1:10101/ with parameters {
    wildcards =     (
        "session/4729C9AE-64C2-4677-B1A4-0EF6A4EAC550/window/handles"
    );
}
```

**Reply by KazuCocoa** [reply]
mac2 driver supports only NATIVE_APP. If the popup is native elements, you could interact with these elements if they were in the result of page_source. If such a popup is not in the page source result, mac2 driver won't be able to find the element. (This is because mac2 driver depends on XCTest framework by Apple. no such elements mean XCTest framework cannot detect them). Then, the only way to interact with the popup is coordinate based click etc.
